### PR TITLE
feat: Manager user logic

### PR DESCRIPTION
**Pull request by yxxhero**
Signed-off-by: yxxhero <aiopsclub@163.com>

## Description
Manager user logic.

## Related Issue
link: https://github.com/dragonflyoss/Dragonfly2/issues/376

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist:

- [x] Code compiles correctly.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: manager/server/router.go [hunk 1]
```diff
@@ -58,12 +63,21 @@ func initRouter(verbose bool, service service.REST) (*gin.Engine, error) {
...
 	ci := apiv1.Group("/cdns")
+	ci.Use(jwt.MiddlewareFunc())
 	ci.POST("", h.CreateCDN)
 	ci.DELETE(":id", h.DestroyCDN)
 	ci.PATCH(":id", h.UpdateCDN)
 	ci.GET(":id", h.GetCDN)
 	ci.GET("", h.GetCDNs)
 
+	// Auth
+	ai := apiv1.Group("/auth")
```
Comment (gaius-qi) [inline manager/server/router.go:74]: Use path `api/v1/user`.

File: manager/service/user.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+}
+
+func (s *rest) Register(json types.RegisterRequest) (*model.User, error) {
+	passwordSalt, err := digestutils.GenerateRandomSalt(16)
+	if err != nil {
+		return nil, err
+	}
+
+	avatar := ""
```
Comment (gaius-qi) [inline manager/service/user.go:32]: Remove setting the default avatar, fill the default avatar on the frontend.

File: manager/types/user.go [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+type LoginRequest struct {
+	Name     string `form:"name" binding:"required"`
+	Password string `form:"password" binding:"required"`
+}
+
+type RegisterRequest struct {
+	Name     string `form:"name" binding:"required"`
+	Password string `form:"password" binding:"required"`
+	Email    string `form:"email" binding:"required"`
+	Phone    string `form:"phone" binding:"required"`
```
Comment (gaius-qi) [inline manager/types/user.go:12]: User `omitempty`.

File: manager/service/user.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+	if len(json.Avatar) == 0 {
+		avatar = defaultAvatar
+	} else {
+		avatar = json.Avatar
+	}
+
+	encryptedPassword := digestutils.Sha256(json.Password, passwordSalt)
```
Comment (gaius-qi) [inline manager/service/user.go:39]: Use `bcrypt.GenerateFromPassword`.

File: manager/service/user.go [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+)
+
+const defaultAvatar = "http://defaultAvatar"
+
+func (s *rest) Login(json types.LoginRequest) (*model.User, error) {
+	user := model.User{}
+	if err := s.db.Where("name = ?", json.Name).First(&user).Error; err != nil {
+		return nil, err
+	}
+	if digestutils.Sha256(json.Password, user.PasswordSalt) != user.EncryptedPassword {
```
Comment (gaius-qi) [inline manager/service/user.go:18]: Use `bcrypt.CompareHashAndPassword`.

File: manager/model/user.go [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+)
+
+type User struct {
+	Model
+	Email             string `gorm:"column:email;size:256;uniqueIndex;not null" json:"email"`
+	Name              string `gorm:"column:name;size:256;uniqueIndex;not null" json:"name"`
+	EncryptedPassword string `gorm:"column:encrypted_password;size:1024" json:"encrypted_password"`
+	Avatar            string `gorm:"column:avatar;size:256;not null" json:"avatar"`
+	Phone             string `gorm:"column:phone;size:256;not null" json:"phone"`
+	PasswordSalt      string `gorm:"column:password;size:256;not null" json:"password"`
```
Comment (gaius-qi) [inline manager/model/user.go:15]: Remove `PasswordSalt`.
Refer to https://medium.com/@jcox250/password-hash-salt-using-golang-b041dc94cb72

**Review by gaius-qi — CHANGES_REQUESTED** [review]

**Comment by yxxhero** [comment]
@gaius-qi ping

File: manager/model/user.go [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+const (
+	UserStateEnabled  = "enable"
+	UserStateDisabled = "disable"
+)
+
+type User struct {
+	Model
+	Email             string `gorm:"column:email;size:256;uniqueIndex;not null" json:"email"`
+	Name              string `gorm:"column:name;size:256;uniqueIndex;not null" json:"name"`
+	EncryptedPassword string `gorm:"column:encrypted_password;size:1024" json:"encrypted_password"`
```
Comment (gaius-qi) [inline manager/model/user.go:12]: Remove `json:"encrypted_password"`

File: manager/handlers/user.go [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+	userInfo, err := h.Service.Register(json)
+	if err != nil {
+		ctx.Error(err)
+		return
+	}
+	userResponseInfo := *userInfo
+	userResponseInfo.EncryptedPassword = ""
+
+	ctx.JSON(http.StatusOK, userInfo)
```
Comment (gaius-qi) [inline manager/handlers/user.go:35]: Remove user model's `json:"encrypted_password"` and return userInfo directly.

**Review by gaius-qi — DISMISSED** [review]

**Review by gaius-qi — APPROVED** [review]

**Review by garfield009 — APPROVED** [review]

**Pull request merged**